### ISSUE TITLE
How to reproduce lerobot/pi0_libero_finetuned?

### ISSUE THREAD

**Issue by PuzhenYuan** [huggingface/lerobot#2156]
Thanks for the great work!

I evaluated lerobot/pi0_libero_finetuned on libero goal datasets.
When using n_action_steps=50, the success rate is ~ 75%
When using n_action_steps=10, the success rate is ~ 90%

I tried to reproduce the training results, so I mainly refered to [train_config.json](https://huggingface.co/lerobot/pi0_libero_finetuned/blob/main/train_config.json) in the `lerobot/pi0_libero_finetuned` repo, which has one key value pair in the config dict:
```
"pretrained_path": "pepijn223/pi0_libero_finetuned_extra"
```

So I also refered to the [train_config.json](https://huggingface.co/pepijn223/pi0_libero_finetuned_extra/blob/main/train_config.json) in th `pepijn223/pi0_libero_finetuned_extra` repo, which also has the key value pair:
```
"pretrained_path": "lerobot/pi0_libero_finetuned"
```
This again points back to the checkpoint that depends on it.

And my questions are, how are these checkpoints actually trained, and can anyone provide a train_config.json in the latest lerobot version that can reproduce lerobot/pi0_libero_finetuned?

Please also share some successful training configs if possible!

**Comment by jadechoghari** [comment]
Hey @PuzhenYuan the train config is provided here no?: https://huggingface.co/lerobot/pi0_libero_finetuned/blob/main/train_config.json

**Comment by PuzhenYuan** [comment]
@jadechoghari Yes I also see the train_config.json, but it seems that it was pretrained from pepijn223/pi0_libero_finetuned_extra, and pepijn223/pi0_libero_finetuned_extra was pretrained from lerobot/pi0_libero_finetuned, which makes me confused.

**Comment by pkooij** [comment]
Yes I did multiple training runs, in this case. You should be able to reproduce it by doing a longer training run directly from lerobot/pi0_libero_finetuned or even lerobot/pi0_base

**Comment by PuzhenYuan** [comment]
@pkooij Thanks! Could you tell me approximately how many steps you have trained?

**Comment by pkooij** [comment]
I trained for ~10k steps with bs=256. But I have seen other people train with similar batch size for as much as 100k steps

**Comment by aopolin-lv** [comment]
> [@pkooij](https://github.com/pkooij) Thanks! Could you tell me approximately how many steps you have trained?

Hello, I want to know, after you fine-tune pi0 with 10k/100k steps, how much can the libero average reach?

**Comment by fenghao999** [comment]
> I trained for ~10k steps with bs=256. But I have seen other people train with similar batch size for as much as 100k steps

Hi @pkooij , I’m trying to reproduce lerobot/pi0_libero_finetuned_v044 on LIBERO as a step of doing my own research. I’ve been studying the training configs published on Hugging Face for both pepijn223/pi0_libero_finetuned_extra and lerobot/pi0_libero_finetuned_v044, and I’ve also been running my own training experiments. My best result so far is only around 36% success rate on libero_object, which is not acceptable.

I was hoping you could help clarify a few details about the training recipe.

1. Could you confirm the exact multi-stage training pipeline? My current understanding is:

    Stage 1: lerobot/pi0_libero_base → fine tune → pepijn223/pi0_libero_finetuned_extra
    Stage 2: pepijn223/pi0_libero_finetuned_extra → fine tune → lerobot/pi0_libero_finetuned_v044

The reason I’m confused is that, in the posted train_config.json files, pi0_libero_finetuned_extra appears to use lerobot/pi0_libero_finetuned as its pretrained model, while pi0_libero_finetuned_v044 appears to use pi0_libero_finetuned_extra as its pretrained model. That made me wonder whether any Hugging Face repo references became stale over time, or whether there were additional intermediate stages.

2. I have a few questions about the learning rate schedule. Based on my understanding of the two-stage pipeline:

- For pi0_libero_finetuned_extra, the config shows scheduler_warmup_steps=1000 and scheduler_decay_steps=6000 in the policy fields, with steps=6000. Did you explicitly set --policy.scheduler_decay_steps=6000 and --steps=6000 to prevent auto-scaling?

- Similarly, for pi0_libero_finetuned_v044, the config shows scheduler_warmup_steps=1000 and scheduler_decay_steps=10000 in the policy fields. Did you explicitly set --policy.scheduler_decay_steps=10000 and --steps=10000 to prevent auto-scaling?


3. I also wanted to ask about a few other settings:

- During training, did you set --policy.n_action_steps=50 or --policy.n_action_steps=10? 

- Both configs use seed=1000. Was that important for reproducing the reported result, or was it mainly just a convention?

- Have you experimented with changing the batch size? If so, do you have any advice on how to adjust the number of training steps and learning rate accordingly?

Any guidance would be greatly appreciated. Thank you very much for your time.

**Comment by pkooij** [comment]
Hi @fenghao999, for 1. I am not sure, but. I think the HF reference is correct. for 2. yes we explicitly set --policy.scheduler_decay_steps=6000 is steps is 6000. And set --policy.scheduler_decay_steps=10000 if --steps=10000. For 3. We trained with --policy.n_action_steps=50 if I remember correctly but used n_action_steps=10 for rollout. However I think you could set --policy.n_action_steps=10 as libero is recorded at 10fps, so one chunk will equal 1sec as with most policies when doing --policy.n_action_steps=10. For batch size I recommend taking one that's as large as possible given your gpu.

**Comment by fenghao999** [comment]
@pkooij Thank you so much for the information. They are very helpful. 

Yet, I’m still a bit confused by what looks like a circular reference in the public Hugging Face train_config.json files: the pretrained_path for "lerobot/pi0_libero_finetuned_v044" is "pepijn223/pi0_libero_finetuned_extra", while the pretrained_path for "pepijn223/pi0_libero_finetuned_extra" is "lerobot/pi0_libero_finetuned".

My current guess is that "lerobot/pi0_libero_finetuned" may have been an earlier version of "lerobot/pi0_libero_finetuned_v044". Is that correct?

If so, was that earlier lerobot/pi0_libero_finetuned checkpoint trained from lerobot/pi0_libero_base? And could you share how many training steps were used for that stage?
